### PR TITLE
Fix #632: theme rest client fix

### LINKED ISSUE #632: Crash (8.9): NFE in REST client when deserializing theme network response
```
Fatal Exception: java.lang.NumberFormatException: For input string: ",730 ₹"
       at java.lang.Integer.parseInt(Integer.java:510)
       at java.lang.Integer.valueOf(Integer.java:611)
       at org.wordpress.android.fluxc.network.rest.wpcom.theme.ThemeRestClient.createThemeFromWPComResponse(ThemeRestClient.java:265)
       at org.wordpress.android.fluxc.network.rest.wpcom.theme.ThemeRestClient.createThemeListFromArrayResponse(ThemeRestClient.java:305)
       at org.wordpress.android.fluxc.network.rest.wpcom.theme.ThemeRestClient.access$700(ThemeRestClient.java:41)
       at org.wordpress.android.fluxc.network.rest.wpcom.theme.ThemeRestClient$7.onResponse(ThemeRestClient.java:138)
       at org.wordpress.android.fluxc.network.rest.wpcom.theme.ThemeRestClient$7.onResponse(ThemeRestClient.java:133)
       at org.wordpress.android.fluxc.network.rest.GsonRequest.deliverResponse(GsonRequest.java:50)
       at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:99)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:153)
       at android.app.ActivityThread.main(ActivityThread.java:6208)
```

### PR DESCRIPTION
**Pull request by maxme**
IMO the real issue is in the API, we should get 2 fields (currency and price) - so instead of trying to parse the price, I just added a "priceText" field in the ThemeModel

cc @oguzkocer @tonyr59h

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/theme/ThemeRestClient.java [hunk 1]
```diff
@@ -235,9 +235,11 @@ private static ThemeModel createThemeFromWPComResponse(WPComThemeResponse respon
         theme.setScreenshotUrl(response.screenshot);
         theme.setDescription(response.description);
         theme.setDownloadUrl(response.download_uri);
-        if (!TextUtils.isEmpty(response.price)) {
-            theme.setCurrency(response.price.substring(0, 1));
-            theme.setPrice(Integer.valueOf(response.price.substring(1)));
+        if (TextUtils.isEmpty(response.price)) {
+            theme.setFree(true);
+        } else {
+            theme.setFree(false);
```
Comment (oguzkocer) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/theme/ThemeRestClient.java]: Is it at all possible that price field might be populated with `0` indicating that it's free? From what I understand, this isn't the case, but wanted to make sure.

We could possibly add a comment here saying price field is always empty for free themes.

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java [hunk 2]
```diff
@@ -153,6 +153,10 @@ public void onUpgrade(SQLiteDatabase db, WellTableManager helper, int oldVersion
                         + "STYLESHEET TEXT,CURRENCY TEXT,PRICE REAL,ACTIVE INTEGER,AUTO_UPDATE INTEGER,"
                         + "AUTO_UPDATE_TRANSLATION INTEGER,IS_WP_COM_THEME INTEGER)");
                 oldVersion++;
+            case 16:
+                db.execSQL("alter table ThemeModel add FREE integer;");
```
Comment (oguzkocer) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: Shouldn't this be `boolean`?

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java [hunk 3]
```diff
@@ -153,6 +153,10 @@ public void onUpgrade(SQLiteDatabase db, WellTableManager helper, int oldVersion
...
+                db.execSQL("alter table ThemeModel add PRICE_TEXT integer;");
```
Comment (oguzkocer) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: Should we also remove the `PRICE` field now that we don't use it anymore?

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/theme/ThemeRestClient.java (hunk 1 above)
Comment (maxme) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/theme/ThemeRestClient.java]: > Is it at all possible that price field might be populated with 0 indicating that it's free?

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java (hunk 3 above)
Comment (maxme) [reply fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: AFAIK, we can't drop fields in sqlite without copying/filtering the table.

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java (hunk 2 above)
Comment (maxme) [reply fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: There is no boolean types in `sqlite` (we use integer 0/1 instead).

Comment (oguzkocer) [reply fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: We probably should change the previous migration statements in a different PR. I at least have used it once and I think it was also used before.

**Comment by oguzkocer** [comment]
:shipit:

**Pull request merged**